pyoracc/atf/common/atffile.py

```diff
@@ -72,6 +73,32 @@ def __str__(self):
     def serialize(self):
         return AtfFile.template.render_unicode(**vars(self))
 
+    def to_json(self, skip_empty=True, **kwargs):
+        '''Return a JSON representation of the parsed file.
+
+        The optional skip_empty argument determines whether keys
+        with empty values are included in the output. Set it to
+        False to see all possible object members.
+
+        Otherwise it accepts the same optional arguments as
+        json.dumps().'''
+        def _make_serializable(obj):
+            '''Construct a dict representation of an object.
+
+            This is necessary to handle our custom objects
+            which json.JSONEncoder doesn't know how to
+            serialize.'''
+
+            return {k: v
+                    for k, v in vars(obj).items()
+                    if not str(k).startswith('_') and not (
+                        skip_empty and not v and not isinstance(v, bool)
```

Comment:
From a quick look, this would also skip values that are 0, but I'm not sure if that ever comes up?

Reply:
Good point. The only numerical value I see in the sample corpus is the [Ruling](https://github.com/oracc/pyoracc/blob/ecac63d/pyoracc/model/ruling.py#L28) object, which has a `count` property. I don't think that's likely to be zero.

I wanted to pass boolean values because I thought it would be helpful to test them directly. It turns out `bool` is an instance of `numbers.Number`, so we could include 0 just by changing the class we check against.

What do you think?

Reply:
I convinced myself.